### PR TITLE
feat(backend): Users Verify Password API

### PR DESCRIPTION
**Pull request by dvargas92495**
## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [x] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

Adds https://clerk.dev/docs/reference/backend-api/tag/Users#operation/VerifyPassword

### CODE REVIEW

File: packages/backend/src/api/endpoints/UserApi.ts [hunk 1]
```diff
@@ -166,4 +171,15 @@ export class UserAPI extends AbstractAPI {
...
   }
+
+  public async verifyPassword(params: VerifyPasswordParams) {
+    const { userId, password } = params;
+    this.requireId(userId);
+
+    return this.request<{verified: true}>({
+      method: 'POST',
+      path: joinPaths(basePath, userId, 'verify_password'),
+      bodyParams: { userId, password },
```
Comment (dimkl) [inline packages/backend/src/api/endpoints/UserApi.ts:182]: 🔧 based on the [docs](https://clerk.dev/docs/reference/backend-api/tag/Users#operation/VerifyPassword) the userId is not part of the request `body` but is used in the path.

**Review by dimkl — COMMENTED** [review]
Hello @dvargas92495 👋 , thank you for the contribution!
Could you please update the commit message to follow the conventional-commits specification (https://github.com/clerkinc/javascript/blob/main/docs/MONOREPO.md#a-note-on-commit-messages) ?

**Comment by dimkl** [comment]
Similar PR to be used as example: https://github.com/clerkinc/javascript/pull/857

**Comment by dvargas92495** [comment]
@dimkl edited the first commit message, and PR title!

**Review by dimkl — APPROVED** [review]

**Pull request merged**